internal/testutils/reference_addon.go

```diff
@@ -103,6 +109,9 @@ func (r *ReferenceAddonStage) GetImageSet(version string) (*addonsv1alpha1.Addon
 }
 
 func (r *ReferenceAddonStage) GetMetaBundle(version string) (*utils.MetaBundle, error) {
```

Comment:
is there a reason why we have made defined this method on a pointer receiver ?
We can get rid of the deepcopy stuff if we defined this on a normal object.

Reply:
We can also get rid of `metaBundleMutex` if we change the method receiver to a normal object. (`ExtractAndParseAddons` fn has its own internal mutex)

Reply:
The `ReferenceAddonStage` is a singleton that we initialize once by reading yaml manifests. This is to speedup testing, we load the metadata once globally from `internal/testdata/`. Than we generate the `metaBundle` object using deepcopy instead of reading yaml manifests again. Deepcopying is faster than I/O, and allows each call to `GetMetaBundle` to get a fresh, independent `metaBundle`.

Reply:
The `metaBundleMutex` is superseded by the global mutex on `ExtractAndParseAddons`.

@yashvardhan-kukreja please rebase this PR.

Reply:
@sugarraysam What I was thinking is this, 
```
meta := *r.MetaImageSet
if err := meta.PatchWithImageSet(imageSet); err != nil ...
```
But i guess this doesn't actually clone everything as `r.MetaImageSet` has some pointer attributes.
Lets stick to deepcopying